--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -97,7 +97,7 @@ if (NOT vcl_ext_fmt)
 	set(vcl_ext_fmt "vcl_ext_fmt_tgt")
 endif()
 
-if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/src/externals/memory")
+if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/src/externals/memory/CMakeLists.txt")
 	option(FOONATHAN_MEMORY_BUILD_EXAMPLES "" OFF)
 	option(FOONATHAN_MEMORY_BUILD_TESTS "" OFF)
 	add_subdirectory(src/externals/memory EXCLUDE_FROM_ALL)
@@ -105,7 +105,7 @@ if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/src/externals/memory")
 	set_property(TARGET foonathan_memory_node_size_debugger PROPERTY FOLDER 3rd-party)
 endif()
 
-if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/src/externals/json")
+if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/src/externals/json/CMakeLists.txt")
 	option(JSON_BuildTests "" OFF)
 	option(JSON_Install "" OFF)
 	add_subdirectory(src/externals/json)
